Source/JavaScriptCore/wasm/js/WebAssemblyFunction.cpp

```diff
@@ -415,6 +415,10 @@ MacroAssemblerCodePtr<JSEntryPtrTag> WebAssemblyFunction::jsCallEntrypointSlow()
     jit.loadPtr(entrypointLoadLocation(), scratchJSR.payloadGPR());
     jit.call(scratchJSR.payloadGPR(), WasmEntryPtrTag);
 
+    // Restore stack pointer after call
+    // NOTE: we use addPtr to generate lea
+    jit.addPtr(MacroAssembler::TrustedImm32(-static_cast<int32_t>(totalFrameSize)), MacroAssembler::framePointerRegister, MacroAssembler::stackPointerRegister);
```

Comment:
I think I am missing something. Why do we need to do this? Given that this particular jit.call wasn't a tail call, shouldn't the callee clean up the stack pointer? If not, how do we know that we have the correct frame size?